spec/build_spec.lua

```diff
@@ -451,3 +448,434 @@ describe("LuaRocks build tests #integration", function()
+                  libdirs = libdirs
+               }
+            })
+            
+            lfs.rmdir(location .. "/dir1/dir2")
+            lfs.rmdir(location .. "/dir1")
+            lfs.rmdir(location)
+         end
+         
+         it("returns a table of the modules having as location the src directory", function()
+            test_with_location("src")
+         end)
+         
+         it("returns a table of the modules having as location the lua directory", function()
+            test_with_location("lua")
+         end)
+         
+         it("returns as a second argument a table of the copy directories", function()
+            lfs.mkdir("doc")
+            lfs.mkdir("docs")
+            lfs.mkdir("samples")
+            lfs.mkdir("tests")
+            
+            local _, copy_directories = build_builtin.autodetect_modules({}, {}, {})
+            assert.same(copy_directories, { "doc", "docs", "samples", "tests" })
+            
+            lfs.rmdir("doc")
+            lfs.rmdir("docs")
+            lfs.rmdir("samples")
+            lfs.rmdir("tests")
+         end)
+      end)
+      
+      describe("builtin.run", function()
+         local tmpdir
+         local olddir
+         
+         before_each(function()
+            tmpdir = get_tmp_path()
+            olddir = lfs.currentdir()
+            lfs.mkdir(tmpdir)
+            lfs.chdir(tmpdir)
+            fs.change_dir(tmpdir)
+            path.use_tree(lfs.currentdir())
+         end)
+         
+         after_each(function()
+            if olddir then
+               lfs.chdir(olddir)
+               if tmpdir then
+                  lfs.rmdir(tmpdir)
+               end
+            end
+         end)
+         
+         local c_module_rockspec = [[
+            package = "c_module"
+            version = "1.0-1"
+            source = {
+               url = "http://example.com/c_module"
+            }
+            build = {
+               type = "builtin",
+               modules = {
+                  c_module = "c_module.c"
+               }
+            }
+         ]]
+         local c_module_source = [[
+            #include <lua.h>
+            #include <lauxlib.h>
+
+            int luaopen_c_module(lua_State* L) {
+              lua_newtable(L);
+              lua_pushinteger(L, 1);
+              lua_setfield(L, -2, "c_module");
+              return 1;
+            }
+         ]]
+         
+         it("returns false if the rockspec has no build modules and the its format does not support autoextraction", function()
+            write_file("test-1.0-1.rockspec", [[
+               rockspec_format = "1.0"
+               package = "test"
+               version = "1.0-1"
+               source = {
+                  url = "http://example.com/foo/tar.gz"
+               }
+               build = {}
+            ]], finally)
+            
+            local rockspec = fetch.load_local_rockspec("test-1.0-1.rockspec", true)
```

Comment:
instead of writing a file and using `fetch.load_local_rockspec` the test will be more specific (ie, using less code that is not the focus of the test) if instead you do `rockspecs.from_persisted_table("test-1.0-1.rockspec", tbl)` (and run slightly faster due to less disk I/O :) )